Zend/zend_vm_execute.skl

```diff
@@ -16,6 +16,15 @@ ZEND_API void {%EXECUTOR_NAME%}_ex(zend_execute_data *ex)
 	LOAD_OPLINE();
 	ZEND_VM_LOOP_INTERRUPT_CHECK();
 
+#ifdef ZEND_CHECK_STACK_LIMIT
+	if (zend_call_stack_overflowed(EG(stack_limit))) {
```

Comment:
```suggestion
	if (zend_call_stack_overflowed(EG(stack_limit)) && !EG(exception)) {
```
I think... Otherwise we're going to enter a loop if any ZEND_HANDLE_EXCEPTION overload or destructor in leave_helper invokes the executor again.